test/backend.jl

```diff
@@ -0,0 +1,68 @@
+@testset "Backend" begin
+    import ONNX.ProtoBuf: encode, decode, ProtoEncoder, ProtoDecoder
+    import ONNX: load, array
+    import Tar: extract
+    import Umlaut: Tape, play!
+    
+    "filename.pb to julia array"
+    function pb_to_array(filename::String)
+        pb = decode(ProtoDecoder(open(filename)), TensorProto)
+        return array(pb)
+    end
+
+    """
+        outputs(dirname)
+
+    load all the outputs in the testing folder.
+    directory structure is:
+    dir (test_<name>)
+    |--model.onnx
+    |--test_data_set_0
+       |-- input_0.pb
+       |-- ...
+       |-- input_N.pb
+       |-- output_0.pb
+       |-- ...
+       |-- output_N.pb
+    load all output_X.pb and convert to Julia array
+    """
+    function outputs(dirname::String)
+        readdir(dirname*"/test_data_set_0",join=true)|>
+            f->filter(contains("output"),f).|>
+            pb_to_array
+    end
+    """
+        eval_model(dirname)
+
+    run onnx backend model using the data in directory.
+    directory structure is:
+    dir (test_<name>)
+    |--model.onnx
+    |--test_data_set_0
+       |-- input_0.bp
+       |-- ...
+       |-- input_N.bp
+       |-- output_0.bp
+    Loads model.onnx, input_X.pb using ONNX.jl, and run it.
+    """
+    function eval_model(dirname::String)
+        ## load inputs
+        inputs = readdir(dirname*"/test_data_set_0",join=true)|>
+            f->filter(contains("input"),f).|>
+            pb_to_array
+        ## load the model
+        model = load(dirname*"/model.onnx",inputs...)
+        ## run it
+        return play!(model,inputs...)
+    end
+
+    @testset "Nodes" begin
+        prefix = extract("backend_data.tar")*"/data/node/"
```

Comment:
Are these files deleted after the test? If not, we should probably add the while directory to the gitignore.

Reply:
Good point, I though that if it is going to `/tmp/` I don’t have to worry about it, but you are right.
I will fix that.